### PR TITLE
[SPARK-46002][INFRA][FOLLOWUP] Upgrade to Python 3.9 in daily maven test

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Upgrade to Python 3.9 in daily maven test

### Why are the changes needed?
to be consistent with https://github.com/apache/spark/pull/43902

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @LuciferYang

**Review by LuciferYang — APPROVED** [review]

File: .github/workflows/maven_test.yml [hunk 1]
```diff
@@ -152,20 +152,20 @@ jobs:
...
-          python-version: 3.8
+          python-version: '3.9'
           architecture: x64
-      - name: Install Python packages (Python 3.8)
+      - name: Install Python packages (Python 3.9)
         if: (contains(matrix.modules, 'sql#core'))
         run: |
-          python3.8 -m pip install 'numpy>=1.20.0' pyarrow pandas scipy unittest-xml-reporting 'grpcio==1.56.0' 'protobuf==3.20.3'
-          python3.8 -m pip list
+          python3.9 -m pip install 'numpy>=1.20.0' pyarrow pandas scipy unittest-xml-reporting 'grpcio==1.56.0' 'protobuf==3.20.3'
```
Comment (LuciferYang) [inline .github/workflows/maven_test.yml:167]: Do the versions of the modules need to be upgraded simultaneously? like `grpcio`

Comment (zhengruifeng) [reply .github/workflows/maven_test.yml:167]: good point, I think is should be the same

**Comment by zhengruifeng** [comment]
merged to master, thanks @LuciferYang

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM.

**Pull request closed without merging**